daemon/src/main/scala/io/gearpump/cluster/master/AppManager.scala

```diff
@@ -214,6 +213,12 @@ private[cluster] class AppManager(kvService: ActorRef, launcher: AppMasterLaunch
         case AppMasterNonExist =>
           sender ! AppMasterData(AppMasterNonExist)
       }
+
+    case uploadTasks: UploadTaskJar =>
+      uploadTasks.tasks.foreach(uploadedTasks += _ -> uploadTasks.jar)
```

Comment:
I know your point, Stanley and I discussed about it before. We choose to save these info on master side because we hope user don't need to reupload the jar file if the composing failed accidentally.